plugin/src/test/java/io/jenkins/plugins/analysis/core/steps/WarningChecksPublisherITest.java

```diff
@@ -0,0 +1,182 @@
+        assertThat(getAnalysisResult(reference))
+                .hasTotalSize(4)
+                .hasNewSize(0);
+
+        copyMultipleFilesToWorkspaceWithSuffix(project, NEW_CHECK_STYLE_REPORT);
+        Run<?, ?> run = buildSuccessfully(project);
+        assertThat(getAnalysisResult(run))
+                .hasTotalSize(6)
+                .hasNewSize(2);
+
+        WarningChecksPublisher publisher = new WarningChecksPublisher(getResultAction(run));
+        assertThat(publisher.extractChecksDetails())
+                .hasFieldOrPropertyWithValue("detailsURL", Optional.of(getResultAction(run).getAbsoluteUrl()))
+                .usingRecursiveComparison()
+                .ignoringFields("detailsURL")
+                .isEqualTo(createExpectedCheckStyleDetails());
+    }
+
+    @Test
+    public void shouldConcludeChecksAsSuccessWhenQualityGateIsPassed() {
+        FreeStyleProject project = createFreeStyleProjectWithWorkspaceFiles(NEW_CHECK_STYLE_REPORT);
+        enableAndConfigureCheckstyle(project,
+                recorder -> recorder.addQualityGate(10, QualityGateType.TOTAL, QualityGateResult.UNSTABLE));
+
+        Run<?, ?> build = buildSuccessfully(project);
+        WarningChecksPublisher publisher = new WarningChecksPublisher(getResultAction(build));
+
+        assertThat(publisher.extractChecksDetails().getConclusion())
+                .isEqualTo(ChecksConclusion.SUCCESS);
+    }
+
+    @Test
+    public void shouldConcludeChecksAsFailureWhenQualityGateIsFailed() {
+        FreeStyleProject project = createFreeStyleProjectWithWorkspaceFiles(NEW_CHECK_STYLE_REPORT);
+        enableAndConfigureCheckstyle(project,
+                recorder -> recorder.addQualityGate(1, QualityGateType.TOTAL, QualityGateResult.FAILURE));
+
+        Run<?, ?> build = buildWithResult(project, Result.FAILURE);
+        WarningChecksPublisher publisher = new WarningChecksPublisher(getResultAction(build));
+
+        assertThat(publisher.extractChecksDetails().getConclusion())
+                .isEqualTo(ChecksConclusion.FAILURE);
+    }
+
+    @Test
+    public void shouldParseHtmlMessage() {
+        FreeStyleProject project = createFreeStyleProject();
+        enableWarnings(project, new PVSStudio());
+
+        buildSuccessfully(project);
+
+        copySingleFileToWorkspace(project, "PVSReport.xml", "PVSReport.plog");
+        Run<?, ?> run = buildSuccessfully(project);
+
+        WarningChecksPublisher publisher = new WarningChecksPublisher(getResultAction(run));
+        ChecksDetails details = publisher.extractChecksDetails();
+
+        assertThat(details.getOutput().get().getChecksAnnotations())
+                .usingElementComparatorOnFields("message")
+                .containsOnly(new ChecksAnnotationBuilder()
+                        .withMessage("ERROR:\n"
+                                + "Some diagnostic messages may contain incorrect line number.\n"
+                                + "V002:https://www.viva64.com/en/w/v002/")
```

Comment:
Is the idea here to replace a `<a>` with plain text? Or shouldn't this be `[text](url)`?

Reply:
The message field doesn't support markdown